torchao/quantization/quant_api.py

```diff
@@ -401,10 +404,25 @@ def insert_subclass(lin):
     return insert_subclass
 
 def quantize_(
+    # NOT FOR LAND: this is the same across inference and torchao.float8
     model: torch.nn.Module,
+
+    # NOT FOR LAND: this seems to leak implementation details, in the future
+    # can this be named in a more generic way without mentioning the 
+    # implementation? Short term, seems ok to have float8 training use this
+    # evern though the implementation is different from 100% tensor subclass 
+    # based workflows.
     apply_tensor_subclass: Callable[[torch.nn.Module], torch.nn.Module],
+
+    # NOT FOR LAND: this is the same across inference and torchao.float8
     filter_fn: Optional[Callable[[torch.nn.Module, str], bool]] = None,
+
+    # NOT FOR LAND: this seems inference specific, in the future would we 
+    # consider renaming or some other way to make things more clear?
     set_inductor_config: bool = True,
+
+    # NOT FOR LAND: this seems inference specific, in the future would we 
+    # consider renaming or some other way to make things more clear?
     device: Optional[torch.types.Device] = None,
```

Comment:
this one is more about quantizing layer by layer to avoid peak vram consumption in full precision. would training not also benefit from this?

Granted name is maybe not great

Reply:
yeah, if someone has an fp32 model and want to do fp8 training on cuda, but the fp32 model doesn't fit on cuda, i'm confused how

quantize_(model, train_fp8, device="cuda")

isn't what we would want?

is it an issue because everything is supposed to be in the config?

Reply:
usually "convert model to training" does not do anything to the weights, they stay in high precision throughout training.  Therefore, the issues of "quantize_ takes too long" or "quantize_ makes model not fit" do not appear in that context.

it's possible that someone might want to "quantize for training and modify the weights", just not something I see as a common thing

Reply:
yeah but its off by default

Reply:
something like off by default and docblock says "for inference" sgtm